sdk/typescript/src/utils/verify.ts

```diff
@@ -22,21 +23,64 @@ export async function verifyMessage(
 	scope: IntentScope,
 ) {
 	const signature = toSingleSignaturePubkeyPair(serializedSignature);
+
+	if (scope === IntentScope.PersonalMessage) {
+		const messageBytes = messageWithIntent(
+			scope,
+			bcs.ser(['vector', 'u8'], typeof message === 'string' ? fromB64(message) : message).toBytes(),
+		);
+
+		if (
+			verifySignature(
+				blake2b(messageBytes, { dkLen: 32 }),
+				signature.signature,
+				signature.pubKey.toBytes(),
+				signature.signatureScheme,
+			)
+		) {
+			return true;
+		}
+
+		// Fallback for backwards compatibility, old versions of the SDK
+		// did not properly wrap PersonalMessages in a PersonalMessage bcs Struct
+		const unwrappedMessageBytes = messageWithIntent(
+			scope,
+			typeof message === 'string' ? fromB64(message) : message,
+		);
+
+		return verifySignature(
+			blake2b(unwrappedMessageBytes, { dkLen: 32 }),
+			signature.signature,
+			signature.pubKey.toBytes(),
+			signature.signatureScheme,
+		);
+	}
+
 	const messageBytes = messageWithIntent(
 		scope,
 		typeof message === 'string' ? fromB64(message) : message,
 	);
-	const digest = blake2b(messageBytes, { dkLen: 32 });
-	switch (signature.signatureScheme) {
+
+	return verifySignature(
+		blake2b(messageBytes, { dkLen: 32 }),
+		signature.signature,
+		signature.pubKey.toBytes(),
+		signature.signatureScheme,
+	);
+}
+
+function verifySignature(
```

Comment:
Why not just call into the `verify` method on the PK?

Reply:
🤦 yup, thats what this was supposed to do, I think I just had copied this over from the existing verification method, and forgot to clean that up